src/Storages/System/StorageSystemPartsColumns.cpp

```diff
@@ -182,6 +183,8 @@ void StorageSystemPartsColumns::processNextStorage(
                 columns[res_index++]->insert(index_size_in_bytes);
             if (columns_mask[src_index++])
                 columns[res_index++]->insert(index_size_in_allocated_bytes);
+            if (columns_mask[src_index++])
+                columns[res_index++]->insert(queryToString(part->default_codec->getCodecDesc()));
```

Comment:
But now all part columns will have the same default compression codec. But we can specify codecs for columns separately, like:
```
CREATE TABLE compression_codec(
    id UInt64 CODEC(LZ4),
    data String CODEC(ZSTD),
    ddd Date CODEC(NONE),
    somenum Float64 CODEC(ZSTD(2)),
    somestr FixedString(3) CODEC(LZ4HC(7)),
    othernum Int64 CODEC(Delta)
) ENGINE = MergeTree() ORDER BY tuple();
```

Reply:
Ok, I'll fix it later. thank you very much!

Reply:
I have fixed the problem, please help and look again, thank you. @alesapin